app/js/lib/foxbox/services/light.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'Light';
+
+const p = Object.freeze({
+  // Public getter only properties.
+  manufacturer: Symbol('manufacturer'),
+  model: Symbol('model'),
+  name: Symbol('name'),
+
+  // Private getter only properties.
+  adapter: Symbol('adapter'),
+});
+
+export default class LightService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+
+    // Public getter only properties.
+    this[p.manufacturer] = props.properties.manufacturer;
+    this[p.model] = props.properties.model;
+    this[p.name] = props.properties.name;
+
+    // Private getter only properties.
+    this[p.adapter] = props.adapter;
+
+    Object.seal(this);
+  }
+
+  // Public getter only properties.
+  get type() {
+    return TYPE;
+  }
+
+  get manufacturer() {
+    return this[p.manufacturer];
+  }
+
+  get model() {
+    return this[p.model];
+  }
+
+  get name() {
+    return this[p.name];
+  }
+
+  // Public dynamic property getters.
+  isAvailable() {
+    return this.get('available')
+      .then((response) => {
+        for (let id in response) {
+          return response[id].OnOff === 'On';
+        }
+      });
+  }
+
+  isOn() {
+    return this.get('LightOn')
+      .then((response) => {
+        for (let id in response) {
+          return response[id].OnOff === 'On';
+        }
+      });
+  }
+
+  // Public methods.
+  /**
+   * Turn the bulb on or off.
+   *
+   * @param {boolean} on Whether to turn it on (true) or off (false).
+   * @return {Promise}
+   */
+  turn(on) {
+    const value = on ? 'On' : 'Off';
+    return this.set('LightOn', value);
+  }
+
+  /**
+   * Turn on the bulb.
+   *
+   * @return {Promise}
+   */
+  turnOn() {
+    return this.set('LightOn', true);
```

Comment:
__nit:__ maybe it will be better to make ```turn(on)``` as private and call it inside ```turnOn``` and ```turnOff```. Or leave it public if you think it gives more flexibility